src/algorithm/geo/affine_ops.rs

```diff
@@ -97,12 +97,6 @@ impl<O: OffsetSizeTrait> AffineOps<AffineTransform> for MultiPointArray<O> {
     }
 }
 
-impl<O: OffsetSizeTrait> AffineOps<AffineTransform> for GeometryArray<O> {
-    crate::geometry_array_delegate_impl! {
-        fn affine_transform(&self, transform: &AffineTransform) -> Self;
-    }
-}
-
```

Comment:
I think the ideal conversion would be to remove the _enum_ but then implement algorithms on `&dyn GeometryArrayTrait` instead. So it would act similarly to the existing enum, but based on the trait.

So this would be something like
```rs
impl AffineOps<AffineTransform> for &dyn GeometryArrayTrait {}
```

I wouldn't think it would be too hard to update the `geometry_array_delegate_impl` macro to use `.data_type()` and downcast, instead of the current enum handling

Reply:
Yeah. I want to do this but GeometryArrayTrait can not be used as dyn because it has method that returns Self.

Reply:
Right, to finish off https://github.com/geoarrow/geoarrow-rs/issues/236 we want to make it object safe, removing any reference to Self

Reply:
This should be unblocked from https://github.com/geoarrow/geoarrow-rs/pull/289